### PR TITLE
fix: only parse multipv 1 info string

### PR DESCRIPTION
**Pull request by gahtan-syarif**
fix issue https://github.com/Disservin/fast-chess/issues/354

havent tested it, but i think it works :P

### CODE REVIEW

**Comment by Disservin** [comment]
this wont work, people who dont have multipv wont report multipv, if multipv exists in the output then it needs to be 1, if it doesnt then old logic

**Comment by gahtan-syarif** [comment]
> this wont work, people who dont have multipv wont report multipv, if multipv exists in the output then it needs to be 1, if it doesnt then old logic

something like this then perhaps?
```
std::string UciEngine::lastInfoLine() const {
    // iterate backwards over the output and save the first line
    // that contains "info depth" and score and multipv 1
    for (auto it = output_.rbegin(); it != output_.rend(); ++it) {
        if (it->line.find(" multipv 1 ") != std::string::npos){
            if (it->line.find("info") != std::string::npos &&
                it->line.find(" score ") != std::string::npos &&
                it->line.find(" multipv 1 ") != std::string::npos) {
                return it->line;
            }
        } else {
            if (it->line.find("info") != std::string::npos &&
                it->line.find(" score ") != std::string::npos) {
                return it->line;
            }
        }
    }

    return {};
}

```

**Review by Disservin — COMMENTED** [review]
I think this can be just
```cpp
    for (auto it = output_.rbegin(); it != output_.rend(); ++it) {
        if (it->line.find("info") != std::string::npos &&
            it->line.find(" score ") != std::string::npos &&
            (it->line.find(" multipv ") == std::string::npos ||
             it->line.find(" multipv 1") != std::string::npos)) {
            return it->line;
        }
    }
```
?

**Comment by gahtan-syarif** [comment]
> I think this can be just
> 
> ```c++
>     for (auto it = output_.rbegin(); it != output_.rend(); ++it) {
>         if (it->line.find("info") != std::string::npos &&
>             it->line.find(" score ") != std::string::npos &&
>             (it->line.find(" multipv ") == std::string::npos ||
>              it->line.find(" multipv 1") != std::string::npos)) {
>             return it->line;
>         }
>     }
> ```
> 
> ?

wouldnt this return the first multipv it finds? even if its multipv 2?

**Comment by gahtan-syarif** [comment]
ahh wait, yeah i didnt see the ==

**Comment by gahtan-syarif** [comment]
updated

File: src/engine/uci_engine.cpp [hunk 1]
```diff
@@ -96,10 +96,12 @@ void UciEngine::writeLog() const {
 
 std::string UciEngine::lastInfoLine() const {
     // iterate backwards over the output and save the first line
-    // that contains "info depth" and score
+    // that contains "info depth" and score and multipv 1
```
Comment (Disservin) [inline src/engine/uci_engine.cpp:99]: well this is wrong? it doesnt have to contain multipv 1..

Comment (gahtan-syarif) [reply src/engine/uci_engine.cpp]: fixed 👍

**Pull request merged**